pkg/apply/merge.go

```diff
@@ -239,6 +243,16 @@ func mergeConfigMapForUpdate(current, updated *uns.Unstructured) error {
 	}
 
 	var mergedConfigMap configMapData
+
+	for i, a1 := range st1.AddressPools {
+		for j, a2 := range st2.AddressPools {
+			if a2.Name == a1.Name {
+				st1.AddressPools[i] = a2
+				st2.AddressPools = removeAddressPool(st2.AddressPools, j)
```

Comment:
This may cause a panic since the initial range takes into account the original length. See https://vbauerster.github.io/2017/04/removing-items-from-a-slice-while-iterating-in-go/